### PR TITLE
create endpoint to fix our collections

### PR DESCRIPTION
**Pull request by jamesgorrie**
The basic path is:
* Map over nodes, to create nodes with the right data
* Turn them into a list
* Get the list of futures from Dynamo
* Resolve and return

You have to do the whole tree at once as we rely on the relationships.

### CODE REVIEW

File: collections/conf/routes [hunk 1]
```diff
@@ -10,6 +10,7 @@ GET     /collections                                    controllers.CollectionsC
 POST    /collections                                    controllers.CollectionsController.addChildToRoot
 POST    /collections/*collection                        controllers.CollectionsController.addChildToCollection(collection: String)
 DELETE  /collections/*collection                        controllers.CollectionsController.removeCollection(collection: String)
+GET     /corrected-collections                          controllers.CollectionsController.correctedCollections
```
Comment (theefer) [inline collections/conf/routes]: I know it's just a temporary route, but `POST` would be better (`GET` isn't supposed to have side-effects)

File: collections/app/controllers/CollectionsController.scala [hunk 2]
```diff
@@ -127,9 +151,10 @@ object CollectionsController extends Controller with ArgoHelpers {
           // This is so we don't have to rewrite the Write[Seq[T]]
           (seq => Json.toJson(seq))).contramap(collectionsEntity) ~
       (__ \ "fullPath").write[List[String]] ~
+      (__ \ "correctPath").write[List[String]] ~
```
Comment (jamesgorrie) [inline collections/app/controllers/CollectionsController.scala]: Should probably not be written out.

File: collections/conf/routes (hunk 1 above)
Comment (jamesgorrie) [inline collections/conf/routes]: Good catch.

**Comment by jamesgorrie** [comment]
Removed JSON leaking to client as per discussion with @theefer

**Comment by jamesgorrie** [comment]
Tested PROD data on local and ran in less than a second.

**Comment by jamesgorrie** [comment]
New record from Dynamo looks like
```JSON
{
	"collection": {
		"actionData": {
			"author": "karin.andreasson@guardian.co.uk",
			"date": "2016-01-13T10:59:40.311+00:00"
		},
		"description": "Comics",
		"path": [
			"G2",
			"Comics"
		],
		"pathId": "g2/comics"
	},
	"id": "g2/comics"
}
```

**Comment by jamesgorrie** [comment]
ES response:
```JSON
{
	"collections": [{
		"uri": "https://media-collections.local.dev-gutools.co.uk/images/e74b53fe02f5667bceff5f21d4b25f2539ccc414/G2/Comics",
		"data": {
			"path": [
				"G2",
				"Comics"
			],
			"pathId": "g2/comics",
			"description": "Comics",
			"actionData": {
				"author": "wham.crash@comics.com",
				"date": "2016-01-22T16:33:45.460+00:00"
			}
		},
		"actions": [{
			"name": "remove",
			"href": "https://media-collections.local.dev-gutools.co.uk/images/e74b53fe02f5667bceff5f21d4b25f2539ccc414/G2/Comics",
			"method": "DELETE"
		}]
	}]
}
```

**Comment by jamesgorrie** [comment]
Tested search for old style path with new style collection :+1: (as we use the `pathId` that hasn't changed). This means they might show up lowercase on the image until we migrate those over.

**Comment by theefer** [comment]
Cool! :+1:

**Comment by jamesgorrie** [comment]
I will release and run this when there are less people using the Grid.

**Pull request merged**